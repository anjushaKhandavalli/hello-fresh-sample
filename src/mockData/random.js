const data = {
    "categorys": [
        {
            "menu-items": [
                {
                    "id": "94298",
                    "name": "3 Chicken Wings",
                    "description": "Tender, Spicy and Juicy. Original or Peri-Crusted",
                },
                {
                    "id": "94301",
                    "name": "Chicken Livers and Portuguese Roll",
                    "description": "Chicken Livers Topped with PERi-PERi Sauce, Served with A Roll To Soak Up The Sauce",
                },
                {
                    "id": "94304",
                    "name": "Spicy Mixed Olives (V)",
                    "description": "Co-Starring Garlic, Pepper and Chili",
                },
                {
                    "id": "94307",
                    "name": "Hummus with PERI-PERI Drizzle (V)",
                    "description": "Pour Smoky PERi-PERi Oil Over Creamy Hummus and Dig in with Strips Of Warm Pita",
                },
                {
                    "id": "94310",
                    "name": "Red Pepper Dip (V)",
                    "description": "Dive Into Temping Roasted Red Pepper and Chili Spice Dip with Warm Pita Strips",
                },
                {
                    "id": "94313",
                    "name": "Appeteaser Platter",
                    "description": "Choose any Three Appeteasers",
                },
                {
                    "id": "94316",
                    "name": "All Together Now (V)",
                    "description": "Spicy Mixed Olives, Red Pepper Dip, Hummus with PERi-PERi Drizzle and Warm Pita Strips",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "id": "94334",
                    "name": "PERi-PERi Wedges (P)",
                    "description": "",
                },
                {
                    "id": "94337",
                    "name": "Corn-On-The-Cob",
                    "description": "",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "id": "94349",
                    "name": "1/4 Chicken",
                    "description": "Quarter Chicken Marinated with PERi-PERi Sauce and Grilled",
                },
                {
                    "id": "94352",
                    "name": "1/2 Chicken",
                    "description": "Half Chicken Marinated with PERi-PERi Sauce and Grilled",
                },
                {
                    "id": "94355",
                    "name": "5 Chicken Wings",
                    "description": "Original Flavours or PERi-PEREi Crusted (P)",
                },
                {
                    "id": "94358",
                    "name": "Wing Roulette",
                    "description": "10 Wings in PERi-PERi Flavors",
                },
                {
                    "id": "94361",
                    "name": "Whole Chicken",
                    "description": "Whole Chicken Marinated in PERi-PERi Sauce and Grilled",
                },
                {
                    "id": "94364",
                    "name": "Chicken Butterfly",
                    "description": "Succulent Double Chicken Breast in Crispy Skin",
                },
                {
                    "id": "94367",
                    "name": "Espetada",
                    "description": "Tender Marinated Chicken Thighs and Fresh Peppers, Served on a Skewer",
                },
                {
                    "id": "94370",
                    "name": "Espetada Carnival",
                    "description": "Flame-Grilled PERi-PERi Thighs, Stuffed with Feta and Parsley, Skewered with Mixed Peppers",
                },
                {
                    "id": "94373",
                    "name": "Grilled Chicken Tenders",
                    "description": "Juicy and Tender, Delicious with All Sides",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "id": "94379",
                    "name": "Full Platter",
                    "description": "For 3-4 People, Whole Chicken with  2 Large or 4 Regular Rides",
                },
                {
                    "id": "94382",
                    "name": "No Bones Platter",
                    "description": "For 2-3 People, Chicken Butterfly and a Espetada, with 2 Large or 4 Regular Sides",
                },
                {
                    "id": "94385",
                    "name": "Jumbo Platter",
                    "description": "For 4-6 People, 2 Whole Chickens and 5 Large Sides",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "id": "94399",
                    "name": "Dessert",
                    "description": "",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "id": "94319",
                    "name": "PERi-PERi Chips (P)",
                    "description": "Crunchy Chips with PERi-PERi Sauce",
                },
                {
                    "id": "94322",
                    "name": "Spicy Rice",
                    "description": "Spicy fried rice",
                },
                {
                    "id": "94325",
                    "name": "Coleslaw (M)",
                    "description": "Cabbage, Carrot, Nando&amp;#039;s Special Mayonnaise",
                },
                {
                    "id": "94328",
                    "name": "Grilled Veg",
                    "description": "Grilled Assorted Vegetables",
                },
                {
                    "id": "94331",
                    "name": "Garlic Bread",
                    "description": "Toasted Garlic Flavored Bread",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "id": "94340",
                    "name": "Portuguese Salad",
                    "description": "Fresh Tomato, Nando&amp;#039;s Peri-Olives, Mixed Peppers, Cucumber and Onion on A Bed of Crisp Lettuce",
                },
                {
                    "id": "94343",
                    "name": "Mediterranean Salad",
                    "description": "Piccolo Tomato, Cucumber, Sweet Peppers and Mixed Leaves. Sprinkled with Olives and Feta Cheese",
                },
                {
                    "id": "94346",
                    "name": "Caesar Salad (M)",
                    "description": "Cos Lettuce Sprinkled with Parmesan Cheese, Caesar Dressing and Crunchy Croutons",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "name": "Cataplana Algarve",
                    "description": "PERi-PERi Chicken Thighs, Mixed Peppers, Chickpeas, Spicy Rice and Cataplana Sauce",
                }
            ]
        },
        {
            "menu-items": [
                {
                    "id": "94388",
                    "name": "Chicken Wrap (Y)",
                    "description": "A Lightly Toasted Wrap, Generously Filled With Tender Chicken, Chili Jam, Green Leaf Lettuce and Tangy Yoghurt Sauce",
                },
                {
                    "id": "94391",
                    "name": "Chicken Breast Burger (M)",
                    "description": "Served on A Toasted Portuguese Roll with Fresh Rocket, Tomato, Pickled Red Onions and Perinaise",
                },
                {
                    "id": "94394",
                    "name": "Chicken Caesar Wrap (M)",
                    "description": "Served in A Toasted Wrap Full Of PERi-PERi Chicken, Crunchy Croutons, Parmesan Cheese and Sundried Tomatoes, All Smothered in Creamy Caesar Dressing",
                },
                {
                    "id": "94397",
                    "name": "Chicken Pita (M)",
                    "description": "Served in A Lightly Toasted Pita, Stuffed With Chicken Tenders, Crisp Veggie Salad and Creamy Whip",
                }
            ]
        }
    ],
}