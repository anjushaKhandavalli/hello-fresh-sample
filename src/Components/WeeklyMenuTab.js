import * as PropTypes from "prop-types";
import '../App.css';
import {foodItems} from "../mockData/Receipes";

const weeks = {
    week1: {
        id: "week1",
        dates: "Oct 09 - Oct 15",
        foodItemsIndexStartWith: 0
    },
    week2: {
        id: "week2",
        dates: "Oct 16 - Oct 22",
        foodItemsIndexStartWith: 9
    },
    week3: {
        id: "week3",
        dates: "Oct 23 - Oct 29",
        foodItemsIndexStartWith: 18
    },
    week4: {
        id: "week4",
        dates: "Nov 06 - Nov 12",
        foodItemsIndexStartWith: 27
    },
    week5: {
        id: "week5",
        dates: "Nov 13 - Nov 19",
        foodItemsIndexStartWith: 36
    },
}

const WeeklyMenuTab = ({weekSpan, setWeekSpan, setItems}) => {

    const handleChange = (event) => {
        const weekId = event.target.value
        console.log("weekId", weekId)
        const weekSpanDates = weeks[weekId].dates
        setWeekSpan(weekSpanDates);

        const foodItemsStartIndex = weeks[weekId].foodItemsIndexStartWith
        setItems(foodItems.slice(foodItemsStartIndex, foodItemsStartIndex + 9));
    };

    return (
        <div className="menuHeader">
            <p className="menuFor">{weekSpan ? `Menu for ${weekSpan}` : `Please select the week to show foodItems`}</p>
            <div className="menuButtons">
                <button onClick={handleChange} value={weeks.week1.id}>{weeks.week1.dates}</button>
                <button onClick={handleChange} value={weeks.week2.id}>{weeks.week2.dates}</button>
                <button onClick={handleChange} value={weeks.week3.id}>{weeks.week3.dates}</button>
                <button onClick={handleChange} value={weeks.week4.id}>{weeks.week4.dates}</button>
                <button onClick={handleChange} value={weeks.week5.id}>{weeks.week5.dates}</button>
            </div>
        </div>
    );
}

WeeklyMenuTab.propTypes = {
    weekSpan: PropTypes.string,
    onClick: PropTypes.func,
    weeks: PropTypes.shape({
        week1: PropTypes.shape({
            foodItemsIndexStartWith: PropTypes.number,
            dates: PropTypes.string,
            id: PropTypes.string
        }),
        week2: PropTypes.shape({
            foodItemsIndexStartWith: PropTypes.number,
            dates: PropTypes.string,
            id: PropTypes.string
        }),
        week3: PropTypes.shape({
            foodItemsIndexStartWith: PropTypes.number,
            dates: PropTypes.string,
            id: PropTypes.string
        }),
        week4: PropTypes.shape({
            foodItemsIndexStartWith: PropTypes.number,
            dates: PropTypes.string,
            id: PropTypes.string
        }),
        week5: PropTypes.shape({
            foodItemsIndexStartWith: PropTypes.number,
            dates: PropTypes.string,
            id: PropTypes.string
        })
    })
};

export default WeeklyMenuTab;