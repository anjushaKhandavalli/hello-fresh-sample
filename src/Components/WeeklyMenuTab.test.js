import {fireEvent, getByText, render} from "@testing-library/react";
import WeeklyMenuTab from "./WeeklyMenuTab";
import {foodItems} from "../mockData/Receipes";

describe("WeeklyMenuTab", () => {
    it("should render `Please select the week to show foodItems` when there is no weekSpan", () => {
        const { getByText } = render(<WeeklyMenuTab weekSpan=''/>)

        expect(getByText('Please select the week to show foodItems')).toBeInTheDocument();
    })

    it("should render menu for week text when there is weekSpan", () => {
        const { getByText } = render(<WeeklyMenuTab weekSpan='Oct 16 - Oct 22'/>)

        expect(getByText('Menu for Oct 16 - Oct 22')).toBeInTheDocument();
    })

    it("should render buttons for 5 weeks", () => {
        const component = render(<WeeklyMenuTab weekSpan=''/>)

        expect(component).toMatchSnapshot();
    })

    it("should call setWeekSpan on click of week button with valid weekSpan", () => {
        const mockSetWeekSpan = jest.fn();
        const mockSetItems = jest.fn();
        const { getByTestId } = render(<WeeklyMenuTab weekSpan='Oct 16 - Oct 22' setWeekSpan={mockSetWeekSpan} setItems={mockSetItems}/>)

        fireEvent.click(getByTestId('week2'))

        expect(mockSetWeekSpan).toBeCalledTimes(1);
        expect(mockSetWeekSpan).toBeCalledWith("Oct 23 - Oct 29");
    })

    it("should call setItems on click of week button with valid foodItems", () => {
        const mockSetWeekSpan = jest.fn();
        const mockSetItems = jest.fn();
        const { getByTestId } = render(<WeeklyMenuTab weekSpan='Oct 16 - Oct 22' setWeekSpan={mockSetWeekSpan} setItems={mockSetItems}/>)

        fireEvent.click(getByTestId('week2'))

        expect(mockSetItems).toBeCalledTimes(1);
        expect(mockSetItems).toBeCalledWith(foodItems.slice(9, 18));
    })
})