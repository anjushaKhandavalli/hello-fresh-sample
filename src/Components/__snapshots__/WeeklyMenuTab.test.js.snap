// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyMenuTab should render buttons for 5 weeks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menuHeader"
      >
        <p
          class="menuFor"
          data-testid="menuHeader"
        >
          Please select the week to show foodItems
        </p>
        <div
          class="menuButtons"
        >
          <button
            data-testid="week1"
            value="week1"
          >
            Oct 16 - Oct 22
          </button>
          <button
            data-testid="week2"
            value="week2"
          >
            Oct 23 - Oct 29
          </button>
          <button
            data-testid="week3"
            value="week3"
          >
            Nov 06 - Nov 12
          </button>
          <button
            data-testid="week4"
            value="week4"
          >
            Nov 13 - Nov 19
          </button>
          <button
            data-testid="week5"
            value="week5"
          >
            Nov 20 - Nov 26
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menuHeader"
    >
      <p
        class="menuFor"
        data-testid="menuHeader"
      >
        Please select the week to show foodItems
      </p>
      <div
        class="menuButtons"
      >
        <button
          data-testid="week1"
          value="week1"
        >
          Oct 16 - Oct 22
        </button>
        <button
          data-testid="week2"
          value="week2"
        >
          Oct 23 - Oct 29
        </button>
        <button
          data-testid="week3"
          value="week3"
        >
          Nov 06 - Nov 12
        </button>
        <button
          data-testid="week4"
          value="week4"
        >
          Nov 13 - Nov 19
        </button>
        <button
          data-testid="week5"
          value="week5"
        >
          Nov 20 - Nov 26
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
